sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+                                                                          "Content-Length",
+                                                                          "Content-Type",
+                                                                          "Date",
+                                                                          "ETag",
+                                                                          "Expires",
+                                                                          "If-Match",
+                                                                          "If-Modified-Since",
+                                                                          "If-None-Match",
+                                                                          "If-Unmodified-Since",
+                                                                          "Last-Modified",
+                                                                          "Pragma",
+                                                                          "Request-Id",
+                                                                          "Retry-After",
+                                                                          "Server",
+                                                                          "Transfer-Encoding",
+                                                                          "User-Agent",
+                                                                          "WWW-Authenticate" ];
+    private static readonly HashSet<string> s_defaultAllowedQueryParameters = ["api-version"];
+    private static readonly PipelineMessageSanitizer s_defaultSanitizer = new(s_defaultAllowedQueryParameters, s_defaultAllowedHeaderNames);
+
+    internal const bool DefaultEnableLogging = true;
+    internal const bool DefaultEnableMessageContentLogging = false;
+    internal const int DefaultMessageContentSizeLimitBytes = 4 * 1024;
+    internal const double RequestTooLongSeconds = 3.0; // sec
+
+    /// <summary>
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
+    /// </summary>
+    /// <remarks>If an ILoggerFactory is not provided, logs will be written to Event Source
+    /// instead. If an ILoggerFactory is provided, logs will be written to ILogger only and not
+    /// Event Source.</remarks>
+    /// <value>Defaults to <see cref="NullLoggerFactory"/>.</value>
+    public ILoggerFactory? LoggerFactory
+    {
+        get => _loggerFactory;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggerFactory = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if logging should be enabled in this client pipeline.
+    /// </summary>
+    /// <value>Defaults to <c>null</c>. If <c>null</c>, this value will be treated as <c>true</c>.</value>
+    public bool? EnableLogging
```

Comment:
Why are these bool properties nullable as opposed to defaulting to true?

Reply:
This was Anne's idea for managing change tracking. So that it's easier to determine when shared policies should be used.